### ISSUE TITLE
Render buildings

### ISSUE THREAD

**Issue by adamfranco** [ZeLonewolf/openstreetmap-americana#170]
At high zooms, Americana should render building. Thoughts?

Here are some ideas with plain gray fill:
![Screen Shot 2022-02-17 at 4 48 58 PM](https://user-images.githubusercontent.com/25242/154578396-20721e95-6914-41cb-b6f7-02e9a479b721.png)

Most topo-maps use solid black, but this is a heavy treatment that may be too dark.

**Comment by 1ec5** [comment]
I don’t have very many city-scale maps, but many state- and county-scale maps have city insets that show select landmark buildings identifiable by their shapes, like these maps of downtown Cincinnati that depict the former [Riverfront Stadium](https://en.wikipedia.org/wiki/Riverfront_Stadium) or “Cynergy Field” [sic], which was a bowl stadium:

<img src="https://user-images.githubusercontent.com/1231218/154593264-dacf1257-4ab2-47f9-ae6f-907802a94225.jpg" width="300" alt="Huff Realty (Engels) Hamilton County 1997"> <img src="https://user-images.githubusercontent.com/1231218/154593330-06cbd965-32e4-461a-80b2-7cd93e1c6fc3.jpg" width="300" alt="Universal Cincinnati Area">

Some city-scale maps color-code buildings by their use:

<img src="https://user-images.githubusercontent.com/1231218/154591795-31a4fe27-de6c-4f2b-a83d-1b1cef56acc7.jpg" width="250" alt="American Map NYC Lower Manhattan"> <img src="https://user-images.githubusercontent.com/1231218/154592060-43b84e94-d845-409c-bda0-7046c26c1ad7.jpg" width="250" alt="Rand McNally Manhattan 1991">
<img src="https://user-images.githubusercontent.com/1231218/154593522-a89b7e37-911e-441f-a905-f344050dcbce.jpg" width="300" alt="American Map NYC">

Some maps leave out the building use but label house numbers:

<img width="384" alt="Mapsco Fort Worth 1992 CBD" src="https://user-images.githubusercontent.com/1231218/154592299-d215df23-62ea-477b-8404-431ac66051e9.png">

The more tourist-oriented, the more detailed the buildings:

<img src="https://user-images.githubusercontent.com/1231218/154591956-668ed775-b246-4ad4-8b82-6d1b9fe5947e.jpg" width="300" title="San Diego Balboa Park">

**Comment by zekefarwell** [comment]
I agree that solid black will probably be too heavy looking.  I think a shade of gray just dark enough to contrast with the backround with a very slightly darker border should look good.

I don't think the offset border for psuedo 3D look works terribly well, but if we want that look there's no reason we can't do actual 3D buildings.  Examples of the kind of thing that is possible [from MapTiler](https://www.maptiler.com/maps/#streets//vector/15.62/-73.984015/40.738563):
![Screen Shot 2022-02-17 at 7 57 27 PM](https://user-images.githubusercontent.com/281482/154597432-c81a3852-8576-445e-9ebd-821c93db33ff.jpg)
![Screen Shot 2022-02-17 at 7 57 55 PM](https://user-images.githubusercontent.com/281482/154597436-40d8ebf5-927d-458e-afc8-d8b83c340424.jpg)

I'm not sure if 3D buildings are appropriate for this style necessarily, but maybe.  Anyway, it is cool that it's an option with webGL.

**Comment by 1ec5** [comment]
3D buildings could run up against the road map aesthetic that this style has adhered to so far, but it wouldn’t be so jarring if we bring in other changes at the same zoom level that deliberately depart from that aesthetic.

A really fitting way to convey building heights and shapes would be [axonometric](https://tangrams.readthedocs.io/en/main/Overviews/Cameras-Overview/#isometric-camera) outlines. However, only MapLibre’s Node.js bindings support this perspective.

/ref mapbox/mapbox-gl-native#10064 mapbox/mapbox-gl-native#11882

**Comment by 1ec5** [comment]
It wouldn’t be necessary to fake a 3D appearance using a border: we can add a [`fill-extrusion`](https://maplibre.org/maplibre-gl-js-docs/style-spec/layers/#fill-extrusion) layer with a fixed [`fill-extrusion-height`](https://maplibre.org/maplibre-gl-js-docs/style-spec/layers/#paint-fill-extrusion-fill-extrusion-height). Then we can tweak the [light source](https://maplibre.org/maplibre-gl-js-docs/style-spec/light/) for better contrast.

**Comment by zekefarwell** [comment]
Exactly.  We don't have to opt for full 3D buildings with real world height.  We can just pretend the buildings are really short for a nice little 3D effect.  Could look something like this from above:
![Screen Shot 2022-02-17 at 8 56 54 PM](https://user-images.githubusercontent.com/281482/154603117-4a28f703-7ce5-433a-9866-f8f5cd0a2a8a.png)
